_posts/fr/2018-03-01-calabash-cucumber.md

```diff
@@ -0,0 +1,706 @@
+![appRegister]({{ site.baseurl }}/assets/2018-03-01-calabash-cucumber/SimuRegister.png){:height="500px"}
+![appRegisterError]({{ site.baseurl }}/assets/2018-03-01-calabash-cucumber/SimuRegisterError.png){:height="500px"}
+![appHome2]({{ site.baseurl }}/assets/2018-03-01-calabash-cucumber/SmuHome2.png){:height="500px"}
+
+## Login Validation
+
+```swift
+var formIsValid = false
+
+func validForm(email: String, password: String) -> Bool {
+    if email == "" && password == "" {
+        Message.showError(error: "Please, add a email and a password")
+
+        return false
+    }
+
+    if !self.isValidEmail(email: email) {
+        Message.showError(error: "Please, add a validate email")
+
+        formIsValid = false
+    }
+
+    if password.count < 4 {
+        Message.showError(error: "Please, enter a password sup to 3 char")
+
+        return false
+    }
+
+    if email != "mail@mail.fr" || password != "1234" {
+        Message.showError(error: "Login and password dont match")
+
+        return false
+    }
+
+    if email == "mail@mail.fr" && password == "1234" {
+        formIsValid = true
+    }
+
+    return formIsValid
+}
+
+func isValidEmail(email: String) -> Bool {
+    let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+
+    let emailTest = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
+
+    return emailTest.evaluate(with: email)
+}
+``` 
+
+## Inscription Validation
+
+```swift
+func validForm(email: String, password: String, copyPass: String) -> Bool {
+    if email == "" && password == "" && copyPass == "" {
+        Message.showError(error: "Please, add a email and a password")
+
+        return false
+    }
+
+    if !self.isValidEmail(email: email) {
+        Message.showError(error: "Please, add a validate email")
+    }
+
+    if password.count < 4 {
+        Message.showError(error: "Please, enter a password sup to 3 char")
+
+        return false
+    }
+
+    if password != copyPass {
+        Message.showError(error: "Your password its not same")
+
+        return false
+    }
+
+    return true
+}
+
+func isValidEmail(email: String) -> Bool {
+    let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+
+    let emailTest = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
+
+    return emailTest.evaluate(with: email)
+}
+``` 
+
+## Message
+
+J'ai ajouté à mon projet la librairie `SwiftMessages` pour customiser mes erreurs.
+
+```swift
+struct Message {
+    static func showError(error: String) {
+        var config = SwiftMessages.Config()
+        config.duration = .seconds(seconds: 3)
+        config.presentationContext = .window(windowLevel: UIWindowLevelStatusBar)
+
+        let view = MessageView.viewFromNib(layout: .messageViewIOS8)
+        view.configureTheme(.error, iconStyle: .subtle)
+        view.configureContent(title: "Erreur", body: error)
+
+        SwiftMessages.show(config: config, view: view)
+    }
+}
+```
+
+# Aws farm device
```

Comment:
Ça mérite un pré-requis en tant que lecteur je m'attendais pas à gérer du AWS j'crois (mais j'y connais rien)

Reply:
C'est une sorte de bonus. Tu pense que je dois mettre `# Bonus - Aws farm device` ?

Reply:
Why not.